tests/unit/test_manager.py

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Test nfs manager utils."""
+
+import os
+from dataclasses import dataclass
+from pathlib import Path
+from subprocess import CalledProcessError
+from types import SimpleNamespace
+from typing import List, Optional, Union
+from unittest.mock import patch
+
+import charms.operator_libs_linux.v0.apt as apt
+import charms.operator_libs_linux.v1.systemd as systemd
+import utils.manager as nfs
+from pyfakefs.fake_filesystem_unittest import TestCase
+
+
+@dataclass(frozen=True, kw_only=True)
+class MountParams:
+    """Test parameters for the `mount` operation."""
+
+    endpoint: str
+    mountpoint: Union[str, os.PathLike]
+    options: Optional[List[str]]
+    master_file: Union[str, os.PathLike]
+    master_data: str
+    map_file: Union[str, os.PathLike]
+    map_data: str
+
+
+@patch("charms.operator_libs_linux.v1.systemd.service_reload")
+@patch("subprocess.run")
+class TestNFSManager(TestCase):
+    """Test nfs manager utils."""
+
+    def setUp(self) -> None:
+        self.setUpPyfakefs()
+        self.fs.create_dir("/etc/auto.master.d")
+        self.fs.create_file(
+            "/proc/mounts",
+            contents="""\
+/dev/sda1 / ext4 rw,relatime,discard,errors=remount-ro 0 0
+192.168.1.150:/data /data nfs4 rw,relatime,vers=4.2 0 0
+[ffcc:aabb::10]:/things /things nfs rw 0 0
+nsfs /run/snapd/ns/lxd.mnt nsfs rw 0 0
+/etc/auto.data /data autofs rw,relatime 0 0
+tmpfs /run/lock tmpfs rw,nosuid,nodev,noexec,relatime 0 0
+""",
```

Comment:
Heh - a common issue with multiline strings; whitespace where you don't want it :woozy_face: 

Rather than borking the formatting like this, we can use the `dedent` function from the `textwrap` module to remove the extraneous whitespace:

```python3
from textwrap import dedent  # Sometimes I just import `textwrap` itself.

contents = dedent(
    """
    /dev/sda1 / ext4 rw,relatime,discard,errors=remount-ro 0 0
    192.168.1.150:/data /data nfs4 rw,relatime,vers=4.2 0 0
    [ffcc:aabb::10]:/things /things nfs rw 0 0
    nsfs /run/snapd/ns/lxd.mnt nsfs rw 0 0
    /etc/auto.data /data autofs rw,relatime 0 0
    tmpfs /run/lock tmpfs rw,nosuid,nodev,noexec,relatime 0 0
    """
).strip()
```

`dedent` will remove the extra whitespace on the left, and `strip` will remove the newline characters that you don't want in the created file.